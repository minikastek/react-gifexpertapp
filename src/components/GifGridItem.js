import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ( { title, url } ) => {

    return (
        <div className="card animate__bounceIn">
            <img src={url} alt={title} />
            <p>{title}</p>
            {/* {img.title} */}
        </div>
    )
}

GifGridItem.propTypes={
    title: PropTypes.string.isRequired
}

GifGridItem.propTypes ={
    url: PropTypes.string.isRequired
}