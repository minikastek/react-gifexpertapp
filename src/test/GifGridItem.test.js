import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { GifGridItem } from '../components/GifGridItem'

describe('Pruebas a GifGridItem',()=>{

    const title ='un titulo';
    const url = 'https://localhost/algo.jpg'
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('debe mostrar el componente correcto',()=>{

        
        expect(wrapper).toMatchSnapshot();

    })

    test('debe de tener un parrafo con el titulo',()=>{
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('debe tener la imagen igual al url y alt de los props',()=>{
        const img = wrapper.find('img');
        //console.log(img.props('src'))
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
        
    })

    test('debe de tener animate bounce', () => {
        
        const div=wrapper.find('div');
        const className=div.prop('className');
        //console.log(div.prop('className')) esto hice para probar 
        expect(className.includes('animate__bounceIn')).toBe(true);
    })
    
})

