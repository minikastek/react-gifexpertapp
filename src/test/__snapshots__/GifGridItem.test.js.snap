// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a GifGridItem debe mostrar el componente correcto 1`] = `
<div
  className="card animate__bounceIn"
>
  <img
    alt="un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    un titulo
  </p>
</div>
`;
